optimum/habana/diffusers/pipelines/stable_diffusion_xl/pipeline_stable_diffusion_xl_inpaint.py

```diff
@@ -0,0 +1,1045 @@
+                strength,
+                device,
+                denoising_start=self.denoising_start if denoising_value_valid else None,
+            )
+
+            # check that number of inference steps is not < 1 - as this doesn't make sense
+            if num_inference_steps < 1:
+                raise ValueError(
+                    f"After adjusting the num_inference_steps by strength parameter: {strength}, the number of pipeline"
+                    f"steps is {num_inference_steps} which is < 1 and not appropriate for this pipeline."
+                )
+            # at which timestep to set the initial noise (n.b. 50% if strength is 0.5)
+            latent_timestep = timesteps[:1].repeat(num_prompts * num_images_per_prompt)
+            # create a boolean to check if the strength is set to 1. if so then initialise the latents with pure noise
+            is_strength_max = strength == 1.0
+
+            # 5. Preprocess mask and image
+            if padding_mask_crop is not None:
+                crops_coords = self.mask_processor.get_crop_region(mask_image, width, height, pad=padding_mask_crop)
+                resize_mode = "fill"
+            else:
+                crops_coords = None
+                resize_mode = "default"
+
+            original_image = image
+            init_image = self.image_processor.preprocess(
+                image, height=height, width=width, crops_coords=crops_coords, resize_mode=resize_mode
+            )
+            init_image = init_image.to(dtype=torch.float32)
+
+            mask = self.mask_processor.preprocess(
+                mask_image, height=height, width=width, resize_mode=resize_mode, crops_coords=crops_coords
+            )
+
+            if masked_image_latents is not None:
+                masked_image = masked_image_latents
+            elif init_image.shape[1] == 4:
+                # if images are in latent space, we can't mask it
+                masked_image = None
+            else:
+                masked_image = init_image * (mask < 0.5)
+
+            # 6. Prepare latent variables
+            num_channels_latents = self.vae.config.latent_channels
+            num_channels_unet = self.unet.config.in_channels
+            return_image_latents = num_channels_unet == 4
+
+            add_noise = True if self.denoising_start is None else False
+            latents_outputs = self.prepare_latents(
+                num_prompts * num_images_per_prompt,
+                num_channels_latents,
+                height,
+                width,
+                prompt_embeds.dtype,
+                device,
+                generator,
+                latents,
+                image=init_image,
+                timestep=latent_timestep,
+                is_strength_max=is_strength_max,
+                add_noise=add_noise,
+                return_noise=True,
+                return_image_latents=return_image_latents,
+            )
+            image_latents = None
```

Comment:
Because this variable is used in [line 711]( https://github.com/huggingface/optimum-habana/pull/869/files#diff-f93f3672ed90c9f5fb4f42963f87ebd2844ccd21a2bf16a1a634f11263a81e32R711).
Error logs:
```
[INFO|pipeline_stable_diffusion_xl_inpaint.py:505] 2024-07-15 04:47:58,050 >> 1 prompt(s) received, 12 generation(s) per prompt, 4 sample(s) per batch, 3 total batch(es).
Traceback (most recent call last):
  File "/workspace/optimum-habana/examples/stable-diffusion/text_to_image_generation.py", line 471, in <module>
    main()
  File "/workspace/optimum-habana/examples/stable-diffusion/text_to_image_generation.py", line 440, in main
    outputs = pipeline(prompt=args.prompts, **kwargs_call)
  File "/usr/local/lib/python3.10/dist-packages/torch/utils/_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "/workspace/optimum-habana/optimum/habana/diffusers/pipelines/stable_diffusion_xl/pipeline_stable_diffusion_xl_inpaint.py", line 711, in __call__
    image_latents_batches, _ = self._split_and_cat_tensors(batch_size, image_latents)
UnboundLocalError: local variable 'image_latents' referenced before assignment
Internal Error: Received signal - Segmentation fault

```